web/src/popover_menus.js

```diff
@@ -272,6 +280,113 @@ export function initialize() {
         },
     });
 
+    tippy_no_propagation("#stream_filters .topic-sidebar-menu-icon", {
+        ...left_sidebar_tippy_options,
+        onShow(instance) {
+            popover_instances.topics_menu = instance;
+            on_show_prep(instance);
+            const elt = $(instance.reference).closest(".topic-sidebar-menu-icon").expectOne()[0];
+            const args = popover_menus_data.get_topic_popover_content_context(elt);
+            instance.setContent(parse_html(render_topic_sidebar_actions(args)));
+        },
+        onMount(instance) {
+            const $popper = $(instance.popper);
+            const stream_id = stream_popover.elem_to_stream_id(
+                $(instance.reference)
+                    .closest(".topic-sidebar-menu-icon")
+                    .closest(".narrow-filter")
+                    .expectOne(),
+            );
+            const topic_name = $(instance.reference)
+                .closest(".topic-sidebar-menu-icon")
+                .closest("li")
+                .expectOne()
+                .attr("data-topic-name");
+            $popper.one("click", ".sidebar-popover-mute-topic", (e) => {
+                if (!stream_id) {
```

Comment:
I'm wondering if there are any situations where we'd want the click handler to function without a `stream_id`? If that's not the case, I believe it would be best to move this if statement outside of all click handlers. What do you think?

Reply:
I have refactored the code to check the `stream_id` at the `onMount` level so that we do not need to check that condition repeatedly in different click_hanlers.